apps/sample-imdb/components/search.tsx

```diff
@@ -1,15 +1,47 @@
 'use client'
-import { useState } from 'react'
+import { useRouter } from 'next/navigation'
+import { FormEvent } from 'react'
 
-export const Search = () => {
-  const [term, setTerm] = useState<string | undefined>()
+export const Search = ({ term }: { term?: string }) => {
+  const router = useRouter()
 
   return (
-    <input
-      type="search"
-      value={term}
-      onChange={(evt) => setTerm(evt.currentTarget.value)}
-      className="h-[2em] w-[50ch] pl-4 border-2 border-neutral-600 rounded-2xl bg-transparent text-white outline-none focus:transform focus:scale-110 origin-left transition-all focus:border-pink-500"
-    />
+    <form
+      onSubmit={(evt: FormEvent<HTMLFormElement>) => {
+        evt.preventDefault()
+        const value = new FormData(evt.currentTarget).get(
+          'searchField'
+        ) as string
```

Comment:
What is the type of this without the `as string`?

You could maybe narrow the time L16 with `typeof value === "string"` instead of this casting

Reply:
FormData key is `string | string[] | File`
but because it's a search input I felt a bit pointless to make runtime check 🤷

Reply:
Yeah, fair enough